pkg/utils/webhook.go

```diff
@@ -84,3 +91,60 @@ func InjectMountPropagation(runtimeNames []string, pod *corev1.Pod) {
 		}
 	}
 }
+
+// InjectHCFSAddresses inject HCFS addresses into pods according to runtimeNames
+func InjectHCFSAddresses(c client.Client, runtimeNames []string, pod *corev1.Pod) (err error) {
+	namespace := pod.GetNamespace()
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	for _, runtimeName := range runtimeNames {
+		var url string
+		dataset := v1alpha1.Dataset{}
+
+		if err = c.Get(context.TODO(), client.ObjectKey{Name: runtimeName, Namespace: namespace}, &dataset); err != nil {
+			return
+		} else if dataset.Status.HCFSStatus == nil {
```

Comment:
Is there any situation where dataset.Status is nil ? Maybe we should check if dataset.Status is nil.

Reply:
> Is there any situation where dataset.Status is nil ? Maybe we should check if dataset.Status is nil.

dataset.Status is not a pointer